internal/scorer/config.go

```diff
@@ -52,6 +52,11 @@ func (i *Input) UnmarshalYAML(value *yaml.Node) error {
 		return errors.New("field must be set")
 	}
 	*i = Input(*raw)
+
+	if i.Weight <= 0 {
+		return errors.New("weight must be greater than 0")
+	}
+
```

Comment:
nit: please but this under the `raw.Field` check and use `raw.Weight` instead of `i.Weight`

Otherwise this is exactly what I had in mind! thanks :)

Reply:
Took care of it, thanks